src/main/rascal/examples/modules/Rename.rsc

```diff
@@ -93,16 +91,13 @@ bool tryParse(type[&T <: Tree] tp, str s) {
 bool isValidName(moduleId(), str name) = tryParse(#ModuleId, name);
 bool isValidName(structId(), str name) = tryParse(#Id, name);
 
-void renameByTree(checkCandidate(Define d, RenameRequest req), Tree modTree, RenameSolver renamer) {
-    println("Checking <modTree.src.top> for occurrences of \'<d.id>\'");
+bool skipCandidate(set[Define] defs, Tree modTree, Renamer _) {
     // Only if the name of the definition appears in the module, consider it a rename candidate
-    if (/Tree t := modTree, "<t>" == d.id) {
-        renamer.collectTModel(modTree.src.top, renameByModel, rename(d, req));
+    set[str] names = {d.id | d <- defs};
+    if (/Tree t := modTree, "<t>" in names) {
```

Comment:
normally you would pattern match on a more specific thing than all trees, for example `/Name n` or `/Identifier i`